cmd/packer-plugin-check/main.go

```diff
@@ -129,11 +129,12 @@ func discoverAndLoad() error {
 	}
 
 	// TODO: validate correctness of plugins loaded by checking them against the output of the `describe` command.
-	builders, provisioners, postProcessors := config.GetPlugins()
+	builders, provisioners, postProcessors, datasources := config.GetPlugins()
```

Comment:
I don't know if the plugin types will grow anytime soon, but I saw this and thought I will probably get the order wrong at some point. What do you think about having `config.GetPlugins()` return a structure, along with an error? 

The structure would have fields for each of the plugin types and the error would only be non-nil when no plugins could be loaded.

Reply:
Good idea!! I'll leave the error out in this case because it's literally just a getter of private attributes.

Reply:
These fields are part of config, I think we can simply remove the `GetPlugins` call and make those public. ( Actually I made them public for #10304 . 🙂